gengokit/clientarggen/strcarver.go

```diff
@@ -0,0 +1,138 @@
+package clientarggen
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/pkg/errors"
+)
+
+// GenerateCarveFunc returns the string form of go code for a function which
+// will parse a string representing a typed slice. The type of the slice is
+// based on the "GoType" of the input ClientArg struct. For example, if GoType
+// field of the ClientArg is "uint32", then the source code returned would be
+// for a function which could accept a string and marshal that string into a
+// []uint32.
+//
+// Supported types are:
+//
+//     uint32
+//     uint64
+//     int32
+//     int64
+//     float32
+//     float64
+//     bool
+//     string
+func GenerateCarveFunc(m *ClientArg) string {
```

Comment:
You have "carve" functions for the primitive types actually implemented in this PR. Why do you need to generate a string form of a function? The contents of the template are near identical to the functions that are already implemented. Am I missing something, or is there redundancy here?